src/vaadin-overlay.html

```diff
@@ -736,6 +736,7 @@
       static get __attachedInstances() {
         return Array.from(document.body.children)
           .filter(el => el instanceof OverlayElement)
+          .filter(el => !el.hasAttribute('closing'))
```

Comment:
`__attachedInstances` is used in `_exitModalState` to restore pointer events for all overlays from the topmost until the first modal overlay is [reached](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-overlay.html#L818).

In the issue/test, both of the two closing modal overlays considered each other to be the frontmost modal one and thus the actual remaining overlay never got it's pointer-events restored.

The fix excludes "closing" overlays from the `__attachedInstances` so they're not considered to be "attached" while the animation is still active.

Reply:
Alternative fix would be to invoke `_exitModalState` inside the `finishClosing` block.

Reply:
nit: we could use existing `.filter()` call instead of adding another one. Not a blocker.